templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -345,52 +306,140 @@ contents:
     type=internal
     EOF
           nmcli c load ${new_conn_file}
-          echo "Loaded new ovs-if-br-ex connection file: ${new_conn_file}"
+          echo "Loaded new $ovs_interface connection file: ${new_conn_file}"
         else
-          nmcli c add type ovs-interface slave-type ovs-port conn.interface br-ex master ovs-port-br-ex con-name \
-            ovs-if-br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac} \
+          nmcli c add type ovs-interface slave-type ovs-port conn.interface "$bridge_name" master "$ovs_port" con-name \
+            "$ovs_interface" 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac} \
             ipv4.route-metric 100 ipv6.route-metric 100
         fi
       fi
 
       # wait for DHCP to finish, verify connection is up
       counter=0
+      configured=false
+
       while [ $counter -lt 5 ]; do
         sleep 5
         # check if connection is active
-        if nmcli --fields GENERAL.STATE conn show ovs-if-br-ex | grep -i "activated"; then
+        if nmcli --fields GENERAL.STATE conn show "$ovs_interface" | grep -i "activated"; then
           echo "OVS successfully configured"
+          update_nm_conn_files "$bridge_name" "$port_name"
           copy_nm_conn_files
-          ip a show br-ex
+          ip a show "$bridge_name"
           ip route show
           nmcli c show
           configure_driver_options ${iface}
-          exit 0
+          configured=true
+          break
         fi
         counter=$((counter+1))
       done
 
       echo "WARN: OVS did not succesfully activate NM connection. Attempting to bring up connections"
       counter=0
       while [ $counter -lt 5 ]; do
-        if nmcli conn up ovs-if-br-ex; then
+        if nmcli conn up "$ovs_interface"; then
```

Comment:
Why do we `WARN` and do this even though the connection might show as activated above?

Reply:
Good catch, this must be fixed. I'll do this loop only if `! configured`

Reply:
Done